analyzers/tests/SonarAnalyzer.UnitTest/Extensions/AttributeDataExtensionsTest.cs

```diff
@@ -141,6 +142,126 @@ public void TryGetAttributeValue_ObjectConversion(object value, Type expectedTyp
             actualValue.Should().Be(value);
         }
 
+        [DataTestMethod]
+        [DataRow(true)]
+        [DataRow(false)]
+        public void HasAttributeUsageInherited_InheritedSpecified(bool inherited)
+        {
+            var code = $$"""
+                    using System;
+
+                    [AttributeUsage(AttributeTargets.All, Inherited = {{(inherited ? "true" : "false")}})]
+                    public class MyAttribute: Attribute
+                    {
+                    }
+
+                    [My]
+                    public class Program
+                    {
+                    }
+                    """;
+            var snippet = new SnippetCompiler(code);
+            var attribute = snippet.GetTypeSymbol("Program").GetAttributes().Single(x => x.HasName("MyAttribute"));
```

Comment:
Should be easy to extract... also with the optional `bool ignoreErrors = false` parameter for the last usage
```suggestion
            var attribute = CompileAttribute(code);
```